apps/core0/subsys/containers/flist.go

```diff
@@ -93,7 +98,16 @@ func (m *containerManager) flistCreate(cmd *pm.Command) (interface{}, error) {
 	// create flist
 	storage := socat.Resolve(args.Storage)
 
-	_, err := zflist("--archive", archivePath, "--create", srcPath, "--backend", storage)
+	zflistArgs := []string{"--archive", archivePath, "--create", srcPath, "--backend", storage}
+	if args.Token != "" {
+		if args.HubAddress != "" {
+			zflistArgs = []string{"--archive", archivePath, "--create", srcPath, "--backend", storage, "--upload", args.HubAddress, "--token", args.Token}
```

Comment:
Why u override the full list instead of just using append ?
```go
zflistArgs = append(zflistArgs, "--upload", args.HubAddress, "--token", args.Token)
```

Reply:
haaa that's the notation I was looking for, thx